proxy/src/auth/backend/mod.rs

```diff
@@ -280,13 +282,47 @@ async fn auth_quirks(
         return Err(auth::AuthError::ip_address_not_allowed(ctx.peer_addr()));
     }
 
+    // check if a VPC endpoint ID is coming in and if yes, if it's allowed
+    let access_blocks = api.get_block_public_or_vpc_access(ctx, &info).await?;
+    if config.is_vpc_acccess_proxy {
+        if access_blocks.vpc_access_blocked {
+            return Err(AuthError::NetworkNotAllowed);
+        }
+        let extra = ctx.extra();
+        let incoming_vpc_endpoint_id = match extra {
+            None => "".to_string(),
+            Some(ConnectionInfoExtra::Aws { vpce_id }) => {
+                // Convert the vcpe_id to a string
+                match String::from_utf8(vpce_id.to_vec()) {
+                    Ok(s) => s,
+                    Err(_e) => "".to_string(),
+                }
+            }
+            Some(ConnectionInfoExtra::Azure { link_id }) => link_id.to_string(),
+        };
+        if incoming_vpc_endpoint_id == "" {
```

Comment:
Maybe to "return Err(AuthError::MissingEndpointName);" if incoming_vpc_endpoint_id is None in match block?

Reply:
Yes, good idea :), thanks!